File: ui/react/src/action/InsertRows.ts
import TableAction from "./TableAction";
import Action from "./Action";

class InsertRows extends TableAction {
    public rowIndex: number;
    public nRows: number;

    public static deserialize(input: any) {
        const sheetName: string = input.sheetName;
        const tableName: string = input.tableName;
        const rowIndex: number = input.rowIndex;
        const nRows: number = input.nRows;
        return new InsertRows(
            sheetName,
            tableName,
            rowIndex,
            nRows
        );
    }

    constructor(
        sheetName: string,
        tableName: string,
        rowIndex: number,
        nRows: number) {

        super(sheetName, tableName);
        this.rowIndex = rowIndex;
        this.nRows = nRows;
    }

    public wrapper(): Action {
        const action = new Action();
        action.insertRows = this;
        return action;
    }

}

export default InsertRows;

File: core/src/main/java/com/ctrip/ferriswheel/core/action/InsertRows.java
package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.core.intf.Action;

public class InsertRows extends TableAction implements Action {
    private int rowIndex;
    private int nRows;

    public InsertRows() {
    }

    public InsertRows(String sheetName, String tableName, int rowIndex, int nRows) {
        super(sheetName, tableName);
        this.rowIndex = rowIndex;
        this.nRows = nRows;
    }

    public int getRowIndex() {
        return rowIndex;
    }

    public void setRowIndex(int rowIndex) {
        this.rowIndex = rowIndex;
    }

    public int getnRows() {
        return nRows;
    }

    public void setnRows(int nRows) {
        this.nRows = nRows;
    }
}